pyproject.toml

```diff
@@ -0,0 +1,271 @@
+    "Intended Audience :: Education",
+    "Intended Audience :: End Users/Desktop",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: MacOS",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX :: Linux",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+    "Topic :: Education",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Information Analysis",
+    "Topic :: Scientific/Engineering :: Visualization",
+    "Topic :: Scientific/Engineering :: Physics",
+]
+
+[tool.poetry.dependencies]
+python = "^3.8,<3.13"
+click = "^8.1.3"
+pyparsing = "^3.1.0"
+pydantic = "^2.3"
+tzlocal = "^5.0.1"
+pillow = "^10.0.0"
+numpy = "^1.24.4"
+
+[tool.poetry.group.dev.dependencies]
+mypy = "^1.4.1"
+poethepoet = "^0.20.0"
+pytest = "^7.4.0"
+black = "^23.3.0"
+ruff = "^0.0.278"
+jinja2 = "^3.1.2"
+pyyaml = "^6.0"
+tox = "^4.6.3"
+pre-commit = "^3.3.3"
+autoflake = "^2.2.0"
+isort = "^5.12.0"
+pytest-xdist = { extras = ["psutil"], version = "^3.3.1" }
+pytest-cov = "^4.1.0"
+types-tzlocal = "^5.0.1.1"
+ipykernel = "^6.25.0"
+
+[tool.poetry.group.docs]
+optional = true
+
+[tool.poetry.group.docs.dependencies]
+mkdocs = "^1.5.2"
+mkdocs-material = "^9.1.21"
+mkdocstrings = { extras = ["python"], version = "^0.22.0" }
+mkdocs-literate-nav = "^0.6.0"
+mkdocs-macros-plugin = "^1.0.2"
+mkdocs-gen-files = "^0.5.0"
+pygments = "^2.15.1"
+
+[build-system]
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
+
+
+[tool.poe.tasks]
+# -------------------------------------------------------------------------------------
+# git hooks
+install-hooks = [
+    { cmd = "poetry install --sync --no-ansi" },
+    { cmd = "poetry run python -m scripts.install_hooks" },
+    { cmd = "poetry run pre-commit install --install-hooks --overwrite" },
+]
+# -------------
+# hook triggers
+run-code-quality-checks = [{ cmd = "pre-commit run --all-files -v" }]
+# -------------------------------------------------------------------------------------
+
+[tool.isort]
+profile = "black"
+src_paths = ["src", "test"]
+
+[tool.black]
+line-length = 88
+target-version = ['py38']
+extend-exclude = '\.pyi?$'
+
+[tool.ruff]
+select = ["ALL"]
+ignore = [
+    "ANN401",
+    "ANN101",
+    "ANN102",
+    "D203",
+    "D205",
+    "D213",
+    "UP007",
+    "UP006",
+    "TCH001",
+    "D105",   # Adding docstrings in all magic methods doesn't make sense, as usually they are just boilerplate to support operators.
+    "FIX002",
```

Comment:
Add explanations what those do, you will find them [here](https://docs.astral.sh/ruff/rules/).